x-pack/plugin/monitoring/src/test/java/org/elasticsearch/xpack/monitoring/exporter/http/HttpExporterIT.java

```diff
@@ -464,27 +465,37 @@ private void assertMonitorVersionResource(final MockWebServer webServer, final b
 
             assertThat(getRequest.getMethod(), equalTo("GET"));
             assertThat(getRequest.getUri().getPath(), equalTo(pathPrefix + resourcePrefix + resource.v1()));
-            assertMonitorVersionQueryString(resourcePrefix, getRequest.getUri().getQuery());
+            assertMonitorVersionQueryString(getRequest.getUri().getQuery(), Collections.emptyMap());
             assertHeaders(getRequest, customHeaders);
 
             if (alreadyExists == false) {
                 final MockRequest putRequest = webServer.takeRequest();
 
                 assertThat(putRequest.getMethod(), equalTo("PUT"));
                 assertThat(putRequest.getUri().getPath(), equalTo(pathPrefix + resourcePrefix + resource.v1()));
-                assertMonitorVersionQueryString(resourcePrefix, getRequest.getUri().getQuery());
+                Map<String, String> parameters = resourcePrefix.startsWith("/_template")
+                    ? Collections.singletonMap(INCLUDE_TYPE_NAME_PARAMETER, "true")
+                    : Collections.emptyMap();
+                assertMonitorVersionQueryString(putRequest.getUri().getQuery(), parameters);
                 assertThat(putRequest.getBody(), equalTo(resource.v2()));
                 assertHeaders(putRequest, customHeaders);
             }
         }
     }
 
-    private void assertMonitorVersionQueryString(String resourcePrefix, String query) {
-        if (resourcePrefix.startsWith("/_template")) {
-            assertThat(query, equalTo(INCLUDE_TYPE_NAME_PARAMETER + "=true&" + resourceVersionQueryString()));
-        } else {
-            assertThat(query, equalTo(resourceVersionQueryString()));
+    private void assertMonitorVersionQueryString(String query, final Map<String, String> parameters) {
+        List <String> kvParams = new ArrayList<>();
+        for (Map.Entry<String, String> param : parameters.entrySet()) {
+            kvParams.add(param.getKey() + "=" + param.getValue());
+        }
+        String queryString = String.join("&", kvParams);
+
+        String completeQueryString = resourceVersionQueryString();
+        if (queryString.length() > 0) {
+            completeQueryString = queryString + "&" + completeQueryString;
         }
+
+        assertThat(query, equalTo(completeQueryString));
```

Comment:
I believe that this assert will only work on with empty or single params, or maps backed by a TreeMap since the order is relevant when comparing the String representation. 

I would suggest to avoid re-building the string, and run the String result of `getRequest.getUri().getQuery()` (e.g. the first param) through https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/rest/RestUtils.java#L40 to get the corresponding Map and then compare the Maps.